### ISSUE TITLE
VISA: wrong check sceKernelPollSema

### ISSUE THREAD

**Issue by 1bsyl** [libsdl-org/SDL#6698]
@isage 
maybe an error here:
sceKernelPollSema returns a signed int.
and 'res' is an unsigned int tested to be negative.

```
unsigned int res;
  ...
  res = sceKernelPollSema(sem->semid, 1);
  if (res < 0) {
            return SDL_MUTEX_TIMEDOUT;
        }
```

https://github.com/libsdl-org/SDL/blame/main/src/thread/vita/SDL_syssem.c#L88

**Comment by isage** [comment]
That's wrong fix. Just remove unsigned. WaitSema returns signed int too.

**Comment by slouken** [comment]
Fixed in f077c6919, thanks!
